src/main/java/org/folio/service/inventory/InventoryItemStatusSyncService.java

```diff
@@ -0,0 +1,147 @@
+   *
+   * @param poLines the po lines
+   * @param currentStatus the current status to find items
+   * @param newStatus the new status to update
+   * @param requestContext the initial request context
+   * @return future with void result when all items processed
+   */
+  public Future<Void> updateItemStatusesInInventory(List<PoLine> poLines,
+                                                    ItemStatus currentStatus,
+                                                    ItemStatus newStatus,
+                                                    RequestContext requestContext) {
+    if (CollectionUtils.isEmpty(poLines)) {
+      return Future.succeededFuture();
+    }
+    Map<String, List<String>> tenantIdToPoLineIds = getTenantIdsToPoLineIdsMap(poLines);
+    List<Future<Void>> futures = new ArrayList<>();
+    for (Map.Entry<String, List<String>> entry : tenantIdToPoLineIds.entrySet()) {
+      String tenantId = entry.getKey();
+      List<String> poLineIds = entry.getValue();
+      RequestContext updatedContext = RequestContextUtil.createContextWithNewTenantId(requestContext, tenantId);
+      Future<Void> updateItemFeature =
+        GenericCompositeFuture.join(
+          StreamEx.ofSubLists(poLineIds, MAX_IDS_FOR_GET_RQ_15)
+            .map(chunk -> VertxFutureRepeater.repeat(MAX_REPEAT_ON_FAILURE, () -> updateItemStatuses(chunk, currentStatus, newStatus, updatedContext)))
+            .toList())
+          .mapEmpty();
+      futures.add(updateItemFeature);
+    }
+    return GenericCompositeFuture.all(futures)
+      .mapEmpty();
+  }
+
+  private Future<Void> updateItemStatuses(List<String> poLineIds,
+                                          ItemStatus currentStatus,
+                                          ItemStatus newStatus,
+                                          RequestContext requestContext) {
+    return inventoryItemManager.getItemsByPoLineIdsAndStatus(poLineIds, currentStatus.value(), requestContext)
+      .map(items -> updateItemStatuses(items, newStatus))
+      .compose(items -> updateItemsInInventory(items, requestContext))
+      .onFailure(e -> logger.error("Failed to update Inventory item status to: {} for po line ids: {}", newStatus.value(), poLineIds));
+  }
+
+  private Map<String, List<String>> getTenantIdsToPoLineIdsMap(List<PoLine> poLines) {
+    Map<String, List<String>> result = new HashMap<>();
+    for (PoLine poLine : poLines) {
+      for (Location location : poLine.getLocations()) {
+        String tenantId = location.getTenantId();
+        result.computeIfAbsent(tenantId, k -> new ArrayList<>()).add(poLine.getId());
+      }
+    }
+    return result;
+  }
+
+  private List<JsonObject> updateItemStatuses(List<JsonObject> items, ItemStatus status) {
+    items.forEach(item -> updateItemStatus(item, status));
+    return items;
+  }
+
+  private JsonObject updateItemStatus(JsonObject item, ItemStatus itemStatus) {
```

Comment:
**Minor**
Actually we do not create a new item but modify existing one so I'd make method void and update line 66 accordingly.